engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/EmbeddedDeploymentManager.scala

```diff
@@ -121,6 +123,24 @@ class EmbeddedDeploymentManager(
     }
   }
 
+  // We make sure that we don't let deploy a scenario when any parameter editor was modified to dictionary one by AdditionalUIConfigProvider
+  // as that would result in failure during compilation before execution
+  private def ensureAdditionalConfigsDoNotContainDictionaryEditors(deploymentData: DeploymentData): Unit = {
```

Comment:
similar logic is encapsulated in AdditionalDictComponentConfigsExtractor. When it changes, we will have to remember that this place should be modified too. IMO it'd be nice to have it in one place

Reply:
I used extracted method from `AdditionalComponentConfigsForRuntimeExtractor` so the logic is in one place

Reply:
Initially I wanted to limit it just to dicts as other configs should not cause problems on runtime. However in Deployment service we only pass the dictionary ones so it's not a problem. I modified it not to focus just on dicts.